include/flox/core/command.hh

```diff
@@ -119,6 +119,45 @@ struct AttrPathMixin {
 };  /* End struct `AttrPathMixin' */
 
 
+/* -------------------------------------------------------------------------- */
+
+/** @brief Extend a command state blob with registry inputs loaded from "path". */
+struct RegistryFileMixin {
+
+  std::optional<std::filesystem::path> registryPath;
```

Comment:
question/suggestion (nonblocking): I don't fully understand mixins, but is there a reason this needs to be optional? Won't `action()` always set it? If there isn't much reason I'd drop the optional

Reply:
Important context for several of these comments is that @aakropotkin requested that there be a way to set this field outside of argument parsing, which means it needs to start in some default state that we can fill in later i.e. it needs to be optional.

This is why we have a setter (so you can set the field outside of argument parsing). The argument parser just reuses that method.

Reply:
For my own understanding, when would we need to do that? For tests even if it wasn't optional, it would default to `""`

Reply:
Honestly I'm not sure, that's a question for @aakropotkin, but I'd rather eliminate the possibility of this path ever being set to `""` anyway.